packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -91,16 +107,33 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
+        const { paymentData } = payment;
+        const isVaultedFlow =
+            paymentData && isVaultedInstrument(paymentData) && isHostedInstrumentLike(paymentData);
+
+        if (!this.orderId && isVaultedFlow) {
+            this.orderId = await this.createOrder();
+        }
+
         if (!this.orderId) {
             throw new PaymentMethodInvalidError();
         }
 
         try {
+            const paymentPayload =
+                paymentData &&
+                isVaultedInstrument(paymentData) &&
+                isHostedInstrumentLike(paymentData) &&
```

Comment:
Why don't we use `isVaultedFlow` instead of 
```
paymentData &&
isVaultedInstrument(paymentData) &&
isHostedInstrumentLike(paymentData) &&
```
?